pkg/maps/cpumap/map.go

```diff
@@ -0,0 +1,161 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package cpumap
+
+import (
+	"fmt"
+	"log/slog"
+	"unsafe"
+
+	"github.com/cilium/hive/cell"
+
+	"github.com/cilium/cilium/pkg/bpf"
+	dpcfgdef "github.com/cilium/cilium/pkg/datapath/linux/config/defines"
+	"github.com/cilium/cilium/pkg/ebpf"
+)
+
+const (
+	CPUMapName = "cilium_cpu_map"
+)
+
+// cpuMapValue is a representation of the value in an eBPF CPU map.
+// This representation changed in 2020 with the introduction of allowing
+// for eBPF programs to be attached to map entries, allowing for chaining
+// XDP programs together. This use case is not supported as of right now,
+// therefore the field is unused.
+//
+// For more information, see
+// https://lore.kernel.org/bpf/5c9febdf903d810b3415732e5cd98491d7d9067a.1594734381.git.lorenzo@kernel.org/
+type cpuMapValue struct {
+	QSize   uint32             `align:"qsize"`
+	BpfProg struct{ Fd int32 } `align:"bpf_prog"`
+}
+
+// CPUMap is a high level interface for working with the XDP cpumap.
+// There are no user interactions that need to happen on the CPUMap,
+// therefore the interface is empty.
+type CPUMap interface{}
+
+// cpuMap is the internal implementation of the CPUMap interface.
+type cpuMap struct {
+	m   *ebpf.Map
+	cfg Config
+}
+
+// Populate should be called on hive startup to populate the CPUMap
+// with entries for each CPU.
+func (c *cpuMap) Populate() error {
+	slog.Warn("populating cpumap")
```

Comment:
Stray debug comment?

Reply:
Yes thank you, there were a couple in there :facepalm: